File: Algorithm/Program Class/C C++ Python/Class 2/Test02/Test02.c
#include<stdio.h>

int main()
{
    int i;
    int t,a,b,c;
    scanf("%d",&t);
    for(i=0; i<t; i++)
    {
        scanf("%d %d %d",&a,&b,&c);
        if(a*a==b*b+c*c||b*b==a*a+c*c||c*c==a*a+b*b)
            printf("good\n");
        else if(a==b||a==c||b==c)
            printf("perfect\n");
        else
            printf("just a triangle\n");

    }
    return 0;
}

File: Algorithm/Class/Java/Class2/src/com/two/Test02.java
package com.two;

import java.util.Scanner;

public class Test02 {
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);

		int n = scanner.nextInt();
		/*
		 * while(n--) {
		 * error java 没有 1 0 只有布尔变量 true false 
		 * 
		 * }
		 */
		for (int i = 0; i < n; i++) {
			int a, b, c;
			a = scanner.nextInt();
			b = scanner.nextInt();
			c = scanner.nextInt();

			if (a * a + b * b == c * c || a * a + c * c == b * b || c * c + b * b == a * a) {
				System.out.println("good");
			}

			else if (a == b || a == c || b == c) {
				System.out.println("perfect");
			} else {
				System.out.println("just a triangle");
			}
		}

	}

}
